EXILED/Exiled.Events/Patches/Events/Scp939/PlayingFootstep.cs

```diff
@@ -0,0 +1,72 @@
+// -----------------------------------------------------------------------
+// <copyright file="PlayingFootstep.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using Exiled.API.Features;
+
+namespace Exiled.Events.Patches.Events.Scp939
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features.Pools;
+    using Exiled.Events.EventArgs.Scp939;
+    using Exiled.Events.Handlers;
+
+    using HarmonyLib;
+    using PlayerRoles.FirstPersonControl.Thirdperson;
+    using PlayerRoles.PlayableScps.Scp939.Ripples;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="FootstepRippleTrigger.OnFootstepPlayed(AnimatedCharacterModel, float)" />
+    /// to add the <see cref="Scp939.PlayingFootstep" /> event.
+    /// </summary>
+    [HarmonyPatch(typeof(FootstepRippleTrigger), nameof(FootstepRippleTrigger.OnFootstepPlayed))]
+    internal static class PlayingFootstep
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            Label ret = generator.DefineLabel();
+
+            int offset = 2;
+            int index = newInstructions.FindLastIndex(i => i.opcode == OpCodes.Call && i.Calls(Method(typeof(RippleTriggerBase), nameof(RippleTriggerBase.CheckVisibility)))) + offset;

```

Comment:
as i see, offset should be `3`
```suggestion
            int offset = 3;
            int index = newInstructions.FindLastIndex(i => i.opcode == OpCodes.Call && i.Calls(Method(typeof(RippleTriggerBase), nameof(RippleTriggerBase.CheckVisibility)))) + offset;

```

Reply:
<img width="663" alt="image" src="https://github.com/user-attachments/assets/a2b052fb-db8f-4fa2-ba32-163c2b5e986e">

Reply:
also `i.Calls(Method(typeof(RippleTriggerBase), nameof(RippleTriggerBase.CheckVisibility)))` is enough, it already check on `OpCodes.Call` opcode
```suggestion
            int offset = 2;
            int index = newInstructions.FindLastIndex(i => i.Calls(Method(typeof(RippleTriggerBase), nameof(RippleTriggerBase.CheckVisibility)))) + offset;

```

Reply:
> as i see, offset should be `3`

fixed in the update, but if offset 2, which looks still works ? anyway